src/fetch/fetchProducts.ts

```diff
@@ -1,33 +1,71 @@
-import CryptoJs from "crypto-js";
+import { v4 as uuidv4 } from "uuid";
+import CryptoJS from "crypto-js";
+import { Product } from "../components/product-list/ProductList";
 
-
-function generateGuid() {
-  let result, i, j;
-  result = '';
-  for(j=0; j<32; j++) {
-    if( j === 8 || j === 12 || j === 16 || j === 20)
-      result = result + '-';
-    i = Math.floor(Math.random()*16).toString(16).toUpperCase();
-    result = result + i;
-  }
-  return result;
+export interface ProductResp {
+  BoltID: string;
+  Name: string;
+  Description: string;
+  AvailabilityCount: number;
+  Prices: { UnitPriceInCents: number; Currency: string }[];
+  Multimedia: {
+    ImageURLs?: string[];
+  };
+  Identifiers: {
+    MerchantProductID: string;
+    SKU: string;
+  };
+  Properties?: {
+    Name: string;
+    NameID?: number;
+    Value: string;
+    ValueID?: number;
+  }[];
 }
 
+export type Catalog = Array<{
+  Current: ProductResp;
+  Parent: ProductResp;
+  Variants: ProductResp[];
+}>;
 
+export interface FetchProductsResp {
+  Catalog: Catalog;
+}
 
-export async function fetchProducts() {
-  const nonce = generateGuid().toLowerCase();
-  const baseString = ":" + nonce;
-  const stagingSecret = "cer6gnol0kor-DWIS"
+export async function fetchProducts(): Promise<Product[]> {
+  const nonce = uuidv4();
+  const baseString = `:${nonce}`;
+  const stagingSecret = "test_sandbox";
 
-  const signatureBytes = CryptoJs.HmacSHA512(baseString, stagingSecret);
-  const authSig = CryptoJs.enc.Base64.stringify(signatureBytes);
-  console.log(authSig);
+  const signatureBytes = CryptoJS.HmacSHA256(baseString, stagingSecret);
+  console.log(signatureBytes);
+  const authSig = CryptoJS.enc.Base64.stringify(signatureBytes);
   const headers = new Headers();
-  headers.append("X-Publisher-Key", "238a244f214c5635a99bf848dc510191");
+  headers.append("X-Publisher-Key", "fbf27e893e8f512fb39cd752b664af3d");
   headers.append("X-Nonce", nonce);
   headers.append("X-Authorization-Signature", authSig);
 
+  return fetch(
+    "https://api-sandbox.bolt.com/v1/products/catalog?merchant_division_id=LkYmU2HuLkws&limit=5",
```

Comment:
Do we need the "limit=5"? we may not need the pagination since we have the sticky button, wdyt?

Reply:
default limit is 5 if not specified, I increased it to 20